source/lib/cache.js

```diff
@@ -1,97 +1,147 @@
 // @flow
 import {AsyncStorage} from 'react-native'
 import moment from 'moment'
-import {API} from '@frogpond/api'
+import {reportNetworkProblem} from '@frogpond/analytics'
+import delay from 'delay'
 
-type BaseCacheResultType<T> = {
+global.AsyncStorage = AsyncStorage
+
+type _BaseCacheResult<T> = {
 	isExpired: boolean,
 	isCached: boolean,
 	value: ?T,
+	source: ?string,
 }
 
-type CacheResultType<T> = Promise<BaseCacheResultType<T>>
+export type CacheResult<T> = Promise<_BaseCacheResult<T>>
 
 function needsUpdate(time: Date, [count, unit]: [number, string]): boolean {
 	return moment(time).isBefore(moment().subtract(count, unit))
 }
 
-function annotateCacheEntry(stored) {
-	// if nothing's stored, note that it's expired and not cached
-	if (stored === null || stored === undefined) {
-		return {isCached: false, isExpired: true, value: null}
+function annotateCacheEntry({ts, ttl}: {ts: string, ttl: string}) {
+	// handle uncached entries and entries that aren't caches, like the homescreen order
+	if (!ts || !ttl) {
+		return {isCached: true, isExpired: false}
 	}
 
-	// migration from old storage
-	if (
-		!('dateCached' in stored && 'timeToCache' in stored && 'value' in stored)
-	) {
-		return {isCached: true, isExpired: true, value: stored}
+	return {
+		isCached: true,
+		isExpired: needsUpdate(new Date(ts), JSON.parse(ttl)),
 	}
+}
 
-	// handle AsyncStorage entries that aren't caches, like the homescreen order
-	if (!stored.timeToCache) {
-		return {isCached: true, isExpired: false, value: stored.value}
-	}
+/// MARK: Utilities
 
-	const date = new Date(stored.dateCached)
-	const isExpired = needsUpdate(date, stored.timeToCache)
-	return {isCached: true, isExpired, value: stored.value}
+async function setItem(
+	key: string,
+	value: any,
+	{ttl, source}: {ttl?: [number, string], source?: string},
+) {
+	console.log('setItem', {key, value, ttl, source})
+	await AsyncStorage.multiSet([
+		[`fp:${key}:data`, JSON.stringify(value)],
+		[`fp:${key}:ts`, new Date().toUTCString()],
```

Comment:
Any particular reason that `toUTCString` is used here? I prefer ISO strings, personally.

Reply:
Switched to toISOString. I didn't realize that toISOString used UTC, and I also didn't realize that toUTCString returned … not an ISO string.

anyway. it's gone now.